fiftyone/core/stages.py

```diff
@@ -1966,181 +1964,295 @@ def _get_list_field_mongo_filter(filter_arg):
-    def _filter_field(self):
-        raise NotImplementedError("subclasses must implement `_filter_field`")
+    .. note::
 
-    def get_filtered_fields(self, sample_collection, frames=False):
-        list_path, is_frame_field = sample_collection._handle_frame_field(
-            self._filter_field
+        Use :class:`FilterLabels` if you simply want to filter entire
+        :class:`fiftyone.core.labels.Keypoint` objects in a field.
+
+    Examples::
+
+        import fiftyone as fo
+        from fiftyone import ViewField as F
+
+        dataset = fo.Dataset()
+        dataset.add_samples(
+            [
+                fo.Sample(
+                    filepath="/path/to/image1.png",
+                    predictions=fo.Keypoints(
+                        keypoints=[
+                            fo.Keypoint(
+                                label="person",
+                                points=[(0.1, 0.1), (0.1, 0.9), (0.9, 0.9), (0.9, 0.1)],
+                                confidence=[0.7, 0.8, 0.95, 0.99],
+                            )
+                        ]
+                    )
+                ),
+                fo.Sample(filepath="/path/to/image2.png"),
+            ]
         )
 
-        if frames == is_frame_field:
-            return [list_path]
+        dataset.default_skeleton = fo.KeypointSkeleton(
+            labels=["nose", "left eye", "right eye", "left ear", "right ear"],
+            edges=[[0, 1, 2, 0], [0, 3], [0, 4]],
+        )
 
-        return None
+        #
+        # Only include keypoints in the `predictions` field whose
+        # `confidence` is greater than 0.9
+        #
+
+        stage = fo.FilterKeypoints("predictions", filter=F("confidence") > 0.9)
+        view = dataset.add_stage(stage)
+
+        #
+        # Only include keypoints in the `predictions` field with less than
+        # four points
+        #
+
+        stage = fo.FilterKeypoints("predictions", labels=["left eye", "right eye"])
+        view = dataset.add_stage(stage)
+
+    Args:
+        field: the :class:`fiftyone.core.labels.Keypoint` or
+            :class:`fiftyone.core.labels.Keypoints` field to filter
+        filter (None): a :class:`fiftyone.core.expressions.ViewExpression`
+            or `MongoDB expression <https://docs.mongodb.com/manual/meta/aggregation-quick-reference/#aggregation-expressions>`_
+            that returns a boolean, like ``F("confidence") > 0.5`` or
+            ``F("occluded") == False``, to apply elementwise to the
+            specified field, which must be a list of same length as
+            :attr:`fiftyone.core.labels.Keypoint.points`
+        labels (None): a label or iterable of keypoint skeleton labels to keep
+        only_matches (True): whether to only include keypoints/samples with
+            at least one point after filtering (True) or include all
+            keypoints/samples (False)
+    """
+
+    def __init__(self, field, filter=None, labels=None, only_matches=True):
```

Comment:
Are you adding `_new_field` or shall I? That's the only other backend functionality I need for the App.